### PR TITLE
4.get/api/reviews

### PR DESCRIPTION
**Pull request by MPlace1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -38,4 +37,31 @@ describe('/api/categories', () => {
...
+});
+
+describe('/api/reviews', () => {
+    describe('GET', () => {
+        test('should return a array of review objects', () => {
+            return request(app)
+                .get("/api/reviews")
+                .expect(200)
+                .then(({ body }) => {
+                    console.log(body)
```
Comment (dav3rid) [inline __tests__/app.test.js]: Can get rid of the log here

File: models/bg.model.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ exports.fetchCategories = () => {
 exports.fetchReviews = () => {
     return db
         .query(
-            `select * from reviews`
+            `select * from reviews order by created_at desc`
         )
         .then((categories) => {
             return categories.rows
```
Comment (dav3rid) [inline models/bg.model.js]: I would rename this parameter - looks like you've fallen victim to copy paste 😬

File: models/bg.model.js [hunk 3]
```diff
@@ -8,4 +8,14 @@ exports.fetchCategories = () => {
...
         })
+}
+
+exports.fetchReviews = () => {
+    return db
+        .query(
+            `select * from reviews`
```
Comment (dav3rid) [inline models/bg.model.js]: There will certainly need to be a little more logic here

**Review by dav3rid — CHANGES_REQUESTED** [review]
Do your tests pass for this one?
I can see that you assert for a comment_count being but I can't see where this gets applied

File: models/bg.model.js [hunk 4]
```diff
@@ -8,4 +8,14 @@ exports.fetchCategories = () => {
         .then((categories) => {
             return categories.rows
         })
+}
+
+exports.fetchReviews = () => {
+    return db
+        .query(
+            `select reviews.*, comments.review_id as comment_count from reviews join comments on reviews.review_id = comments.review_id order by reviews.created_at desc`
```
Comment (xViolaine) [inline models/bg.model.js]: comment_count is supposed to be the total count of all the comments that correspond with this review_id - you are currently not counting anything, have a look at the COUNT function.

Furthermore, you're using "join" which is essentially an "INNER JOIN". An inner join returns rows when there is a match in both tables. 
This is not the correct choice for the task at hand - have another think about which join we'd want to use here!

**Review by xViolaine — CHANGES_REQUESTED** [review]

**Review by xViolaine — APPROVED** [review]
Good job thinking of the full outer join!

**Pull request merged**